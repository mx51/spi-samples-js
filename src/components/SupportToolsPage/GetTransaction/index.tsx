import { GetTransactionPanel } from './GetTransactionPanel';

export default GetTransactionPanel;
