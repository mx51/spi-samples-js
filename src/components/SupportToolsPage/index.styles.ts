import { makeStyles, Theme, createStyles } from '@material-ui/core/styles';

export default makeStyles((theme: Theme) =>
  createStyles({
    root: {
      padding: theme.spacing(3),
      display: 'flex',
      justifyContent: 'center',
    },
  })
);
