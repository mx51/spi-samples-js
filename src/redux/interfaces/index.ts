import { IProductState } from '../reducers/ProductSlice/interfaces';
import { ITerminalState } from '../reducers/TerminalSlice/interfaces';

export interface ISamplePosState {
  terminals: ITerminalState;
  products: IProductState;
}
