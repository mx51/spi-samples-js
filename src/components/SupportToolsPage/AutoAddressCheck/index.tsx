import React, { useState, useEffect } from 'react';
import {
  Typography,
  FormControl,
  InputLabel,
  MenuItem,
  Select,
  Checkbox,
  FormControlLabel,
  FormGroup,
} from '@material-ui/core';
import { Spi as SpiClient } from '@mx51/spi-client-js';
import { serialNumberFormatter } from '../../../utils/common/helpers';
import { isSerialNumberValid, serialNumberValidatorOnChange } from '../../../utils/validators/validators';
import CustomTextField from '../../CustomTextField';
import useStyles from './index.styles';
import { IFormEventValue } from './interfaces';

// eslint-disable-next-line @typescript-eslint/ban-types
async function getTenantsList(setTenantList: Function) {
  const tenants = await SpiClient.GetAvailableTenants('mx51-support-tool', 'mx51-support-tool', 'AU');
  const defaultTenantList = [
    {
      code: 'wbc',
      name: 'Westpac Presto',
    },
    {
      code: 'till',
      name: 'Till Payments',
    },
  ];

  const tenantList = tenants.Data.length ? tenants.Data : defaultTenantList;
  localStorage.setItem('tenants', JSON.stringify(tenantList));
  setTenantList(tenantList);
}

function webSocketFqdn(webFqdn: any, sn: string, tm: boolean, setWebSocketConnectionFqdn: any) {
  const socket = new WebSocket(`wss://${webFqdn}`, 'spi.2.9.0');

  socket.onopen = (e: any) => {
    socket.send('success');
    setWebSocketConnectionFqdn('Successfully connected to webSocket');
  };

  socket.onerror = (error: any) => {
    setWebSocketConnectionFqdn('Error in connecting to webSocket');
  };
}

async function fetchFqdn(
  sn: string,
  tenant: string,
  tm: boolean,
  setFqdn: any,
  setTimeStampFqdn: any,
  setResult: any,
  setErrorResponse: any,
  setGoogleDns: any,
  setWebSocketConnectionFqdn: any
) {
  const response = await fetch(`https://device-address-api${tm ? '-sb' : ''}.${tenant}.mspenv.io/v1/${sn}/fqdn`, {
    headers: {
      'ASM-MSP-DEVICE-ADDRESS-API-KEY': 'DADDRTESTTOOL',
    },
  });
  let fqdn;
  if (response.ok) {
    const data = await response.json();
    fqdn = data.fqdn;
    setFqdn(data.fqdn);
    setTimeStampFqdn(data.last_updated);
    setResult('success');
    webSocketFqdn(data.fqdn, sn, tm, setWebSocketConnectionFqdn);
  } else {
    const data = await response.json();
    setErrorResponse(data);
    setResult('error');
  }

  if (response.ok) {
    const dnsResponse = await fetch(`https://dns.google/resolve?name=${fqdn}`);
    if (dnsResponse.ok) {
      const data = await dnsResponse.json();
      if (dnsResponse.ok && data.Answer) {
        setGoogleDns(data);
        console.log(data);
      } else {
        setGoogleDns('Error in Google Api');
      }
    }
  }
}

async function fetchIp(
  sn: string,
  tenant: string,
  tm: boolean,
  setIp: any,
  setTimeStampIp: any,
  setResult: any,
  setErrorResponse: any
) {
  const response = await fetch(`https://device-address-api${tm ? '-sb' : ''}.${tenant}.mspenv.io/v1/${sn}/ip`, {
    headers: {
      'ASM-MSP-DEVICE-ADDRESS-API-KEY': 'DADDRTESTTOOL',
    },
  });

  if (response.ok) {
    const data = await response.json();
    setIp(data.ip);
    setTimeStampIp(data.last_updated);
    setResult('success');
    // Todo
    // webSocketIp(data.ip);
  } else {
    const data = await response.json();
    setErrorResponse(data);
    setResult('error');
  }
}

function AutoAddressCheck() {
  const [serialNumber, setSerialNumber] = useState({ isValid: true, value: '' });
  const [fqdn, setFqdn] = useState('');
  const [ip, setIp] = useState('');
  const [timeStampFqdn, setTimeStampFqdn] = useState('');
  const [timeStampIp, setTimeStampIp] = useState('');
  const [selectedTenantCode, setSelectedTenantCode] = useState('');
  const [tenantList, setTenantList] = useState(JSON.parse(window.localStorage.getItem('tenants') || '[]'));
  const [otherTenantCode, setOtherTenantCode] = useState(
    !tenantList.find((tenant: Tenant) => tenant.code === selectedTenantCode) ? selectedTenantCode : ''
  );
  const [testMode, setTestMode] = useState(false);
  const [result, setResult] = useState('');
  const [errorResponse, setErrorResponse] = useState({
    request_id: '',
    error_code: 111,
    error: '',
  });
  const [googleDns, setGoogleDns] = useState({
    Answer: [{ name: '', data: '' }],
  });
  const [webSocketConnectionFqdn, setWebSocketConnectionFqdn] = useState('');
  const classes = useStyles();

  // Todo
  // const [webSocketConnectionIp, setWebSocketConnectionIp] = useState('');

  function fetchResponse(sn: string, tenant: string, tm: boolean) {
    fetchFqdn(
      sn,
      tenant,
      tm,
      setFqdn,
      setTimeStampFqdn,
      setResult,
      setErrorResponse,
      setGoogleDns,
      setWebSocketConnectionFqdn
    );
    fetchIp(sn, tenant, tm, setIp, setTimeStampIp, setResult, setErrorResponse);
  }

  useEffect(() => {
    const params = new URLSearchParams(window.location.search);
    const serialNumberFromUrl = params.get('sn');
    const testModeFromUrl = params.get('qa');
    const tenantFromUrl = params.get('tenant');

    const sn = serialNumberFromUrl === null ? '' : serialNumberFromUrl;
    const tm = testModeFromUrl === 'true';
    setSerialNumber({ ...serialNumber, value: sn });
    setTestMode(tm);

    if (sn !== '' && tenantFromUrl) {
      fetchResponse(sn, tenantFromUrl, tm);
    }

    getTenantsList(setTenantList);
  }, []);

  // Todo
  // function webSocketIp(webIp: any) {
  //   const socket = new WebSocket(`ws://${webIp}`, `spi.${new SpiService()._version}`);
  //   socket.onopen = (e: any) => {
  //     socket.send('success');
  //     setWebSocketConnectionIp('Successfully connected to webSocket');
  //   };
  //   socket.onerror = (error: any) => {
  //     setWebSocketConnectionIp('Error in connecting to webSocket');
  //   };
  // }

  const onOtherTextChange = (event: IFormEventValue) => {
    setSelectedTenantCode('other');
    setOtherTenantCode(event.target.value as string);
  };

  const onSerialNumberChange = (event: IFormEventValue) => {
    const value = serialNumberFormatter(event.target.value as string);
    const isValid = isSerialNumberValid(value);
    setSerialNumber({ isValid, value });
  };

  const onTestModeChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setTestMode(event.target.checked);
  };

  const helperText = serialNumber.isValid ? '' : serialNumberValidatorOnChange(serialNumber.value);

  return (
    <div>
      <Typography className={classes.subHeader}>L2 Support and/or Merchants to test auto address</Typography>
      <FormControl variant="outlined" margin="dense" fullWidth data-test-id="supportTenantForm">
        <InputLabel data-test-id="supportTenantFormLabel">Select a tenant</InputLabel>
        <Select
          className={classes.selecter}
          data-test-id="supportTenantSelectorLabel"
          value={selectedTenantCode}
          onChange={(e: IFormEventValue) => setSelectedTenantCode(e.target.value as string)}
        >
          {tenantList.map((tenant: Tenant) => (
            <MenuItem key={tenant.code} value={tenant.code}>
              {tenant.name}
            </MenuItem>
          ))}
          <MenuItem value="other">Other</MenuItem>
        </Select>
      </FormControl>
      {selectedTenantCode === 'other' && (
        <CustomTextField
          dataTestId="tenantField"
          fullWidth
          label="Other"
          margin="dense"
          onBlur={onOtherTextChange}
          onChange={onOtherTextChange}
          value={otherTenantCode}
          variant="outlined"
        />
      )}
      <CustomTextField
        dataTestId="serialNumberField"
        fullWidth
        label="Serial Number"
        margin="dense"
        error={!serialNumber.isValid}
        helperText={helperText}
        onChange={onSerialNumberChange}
        onBlur={onSerialNumberChange}
        required
        value={serialNumber.value}
        variant="outlined"
      />
      <FormGroup>
        <FormControlLabel control={<Checkbox checked={testMode} onChange={onTestModeChange} />} label="Test" />
      </FormGroup>
    </div>
  );
}

export default AutoAddressCheck;
