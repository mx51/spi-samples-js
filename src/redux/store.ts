import { AnyAction, Reducer, combineReducers, configureStore } from '@reduxjs/toolkit';
import { createMigrate, persistReducer, persistStore } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import hardSet from 'redux-persist/es/stateReconciler/hardSet';
import commonReducer from './reducers/CommonSlice/commonSlice';
import pairFormReducer from './reducers/PairFormSlice/pairFormSlice';
import { preAuthReducer } from './reducers/PreAuth/preAuthSlice';
import productSlice from './reducers/ProductSlice/productSlice';
import selectedTerminalReducer from './reducers/SelectedTerminalSlice/selectedTerminalSlice';
import terminalReducer from './reducers/TerminalSlice/terminalsSlice';
import { IPreAuthState } from './reducers/PreAuth/interfaces';
import { PayAtTableState, payAtTableReducer } from './reducers/PayAtTableSlice/payAtTableSlice';
import { ITerminalState } from './reducers/TerminalSlice/interfaces';

const terminalReducerMigration: Any = {
  0: (state: ITerminalState) => {
    Object.keys(state).forEach((key) => {
      if (state[key] && state[key].deviceAddress) {
        state[key].deviceAddress = state[key].deviceAddress.replace(/^wss?:\/\//, '');
      }
      if (state[key] && state[key].secrets) {
        state[key].secrets = {
          encKey: (state[key].secrets as Any).EncKey,
          hmacKey: (state[key].secrets as Any).HmacKey,
        };
      }
    });
    return state;
  },
};

const rootReducer = combineReducers({
  common: commonReducer,
  pairForm: pairFormReducer,
  products: productSlice,
  terminals: persistReducer<ITerminalState, AnyAction>(
    {
      key: 'terminals',
      version: 0,
      storage,
      stateReconciler: hardSet,
      migrate: createMigrate(terminalReducerMigration, { debug: false }),
    },
    terminalReducer
  ) as unknown as Reducer<ITerminalState>,
  selectedTerminal: selectedTerminalReducer,
  preAuth: persistReducer<IPreAuthState, AnyAction>(
    {
      key: 'root',
      storage,
      stateReconciler: hardSet,
    },
    preAuthReducer
  ) as unknown as Reducer<IPreAuthState>,
  payAtTable: persistReducer<PayAtTableState, AnyAction>(
    {
      key: 'payAtTable',
      storage,
      stateReconciler: hardSet,
    },
    payAtTableReducer
  ) as unknown as Reducer<PayAtTableState>,
});

export const store = configureStore({
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
  reducer: rootReducer,
  devTools: process.env.NODE_ENV === 'development',
});

export const persistor = persistStore(store);

export type RootState = ReturnType<typeof rootReducer>;
export type AppStore = typeof store;
export type AppDispatch = typeof store.dispatch;
