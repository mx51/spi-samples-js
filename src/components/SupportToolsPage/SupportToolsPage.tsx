import React from 'react';
import { Container } from '@material-ui/core';
import Layout from '../Layout';
import AutoAddressCheck from './AutoAddressCheck';
import useStyles from './index.styles';

const SupportToolsPage: React.FC = () => {
  const classes = useStyles();

  return (
    <Layout>
      <Container className={classes.root}>
        <AutoAddressCheck />
      </Container>
    </Layout>
  );
};

export default SupportToolsPage;
