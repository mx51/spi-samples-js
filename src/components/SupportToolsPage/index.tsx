import React from 'react';
import { Grid, Typography, Container } from '@material-ui/core';
import Layout from '../Layout';
import AutoAddressCheck from './AutoAddressCheck';
import useStyles from './index.styles';

const SupportToolsPage: React.FC = () => {
  const classes = useStyles();

  return (
    <Layout>
      <Container maxWidth="md" className={classes.root}>
        <Grid container>
          <Grid item xs={8}>
            <Typography variant="h6" component="h1">
              Support Tools
            </Typography>
            <Typography className={classes.title}>Auto Address Check</Typography>
            <AutoAddressCheck />
          </Grid>
        </Grid>
      </Container>
    </Layout>
  );
};

export default SupportToolsPage;
