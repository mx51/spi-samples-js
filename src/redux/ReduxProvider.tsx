import React from 'react';
import { Provider } from 'react-redux';
// store
import { store } from './store';

interface ReduxProviderInterface {
  children: React.ReactNode;
  reduxStore: typeof store;
}

function ReduxProvider({ children, reduxStore }: ReduxProviderInterface): React.ReactElement {
  return <Provider store={reduxStore}>{children}</Provider>;
}

export default ReduxProvider;
