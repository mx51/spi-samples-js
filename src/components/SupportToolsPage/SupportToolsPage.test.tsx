import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';

import '@testing-library/jest-dom/extend-expect';
import SupportToolsPage from '.';
import ReduxProvider from '../../redux/ReduxProvider';
import { store } from '../../redux/store';

describe('Test <SupportToolsPage />', () => {
  test('snapshot for SupportToolsPage component', () => {
    // Act
    const { container } = render(
      <ReduxProvider reduxStore={store}>
        <Router>
          <SupportToolsPage />
        </Router>
      </ReduxProvider>
    );
    // Assert
    expect(container).toMatchSnapshot();
  });
});
