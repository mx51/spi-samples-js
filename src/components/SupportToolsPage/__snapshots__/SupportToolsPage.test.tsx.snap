// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <SupportToolsPage /> snapshot for SupportToolsPage component 1`] = `
<div>
  <div>
    <div
      class="makeStyles-root-5"
    >
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-navbar-6 mui-fixed MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-dense MuiToolbar-gutters"
        >
          <div
            class="MuiBox-root MuiBox-root-11 makeStyles-navbar-8"
          >
            <div
              class="MuiBox-root MuiBox-root-12"
            >
              <button
                aria-label="menu"
                class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-9 MuiIconButton-colorInherit MuiIconButton-edgeStart"
                data-test-id="navbarMenuIcon"
                name="menuIcon"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <svg
                height="15"
                width="48"
              >
                PrimaryLogoIcon.svg
              </svg>
            </div>
          </div>
        </div>
      </header>
    </div>
    <div
      class="makeStyles-root-1"
    >
      <div
        class="MuiTabs-root makeStyles-tabs-2"
        id="supportTabs"
      >
        <div
          class="MuiTabs-scroller MuiTabs-fixed"
          style="overflow: hidden;"
        >
          <div
            aria-label="support tabs"
            class="MuiTabs-flexContainer"
            role="tablist"
          >
            <button
              aria-selected="true"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-selected"
              id="autoAddressTab"
              role="tab"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiTab-wrapper"
              >
                Auto Address Check
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
              id="getTransactionTab"
              role="tab"
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiTab-wrapper"
              >
                Get Transaction
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <span
            class="PrivateTabIndicator-root-13 PrivateTabIndicator-colorSecondary-15 MuiTabs-indicator"
            style="left: 0px; width: 0px;"
          />
        </div>
      </div>
      <div
        aria-labelledby="simple-tab-0"
        id="simple-tabpanel-0"
        role="tabpanel"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg"
        >
          <div
            class="MuiGrid-root makeStyles-container-18 MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <div
                class="MuiBox-root MuiBox-root-40 makeStyles-lookupContainer-21"
              >
                <h1
                  class="MuiTypography-root makeStyles-h1-22 MuiTypography-body1"
                >
                  Auto address check
                </h1>
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  This support tool can be used by Merchants or L2 support users to test Auto address
                </p>
                <hr
                  class="MuiDivider-root makeStyles-divider-24"
                />
                <div
                  class="MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
                    >
                      <h2
                        class="MuiTypography-root makeStyles-h2-43 MuiTypography-body1"
                      >
                        Test Auto address
                      </h2>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7"
                    >
                      <div
                        class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth"
                        data-test-id="autoAddressTenantForm"
                      >
                        <label
                          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
                          data-shrink="false"
                          data-test-id="autoAddressTenantFormLabel"
                        >
                          Tenant
                        </label>
                        <div
                          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
                          data-test-id="autoAddressTenantSelectorLabel"
                        >
                          <div
                            aria-haspopup="listbox"
                            aria-labelledby="autoAddressTenantDropDownLabel"
                            class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                            role="button"
                            tabindex="0"
                          >
                            <span>
                              ​
                            </span>
                          </div>
                          <input
                            aria-hidden="true"
                            class="MuiSelect-nativeInput"
                            tabindex="-1"
                            value=""
                          />
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M7 10l5 5 5-5z"
                            />
                          </svg>
                          <fieldset
                            aria-hidden="true"
                            class="PrivateNotchedOutline-root-56 MuiOutlinedInput-notchedOutline"
                          >
                            <legend
                              class="PrivateNotchedOutline-legendLabelled-58"
                            >
                              <span>
                                Tenant
                              </span>
                            </legend>
                          </fieldset>
                        </div>
                      </div>
                      <p
                        class="MuiTypography-root makeStyles-infoText-45 MuiTypography-body1"
                      >
                        Select the tenant for the Auto address test
                      </p>
                      <div
                        class="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-60 MuiFormControl-marginDense MuiFormControl-fullWidth"
                        data-test-id="autoAddressSerialNumberField"
                        theme="[object Object]"
                      >
                        <label
                          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined Mui-required Mui-required"
                          data-shrink="false"
                        >
                          Serial Number
                          <span
                            aria-hidden="true"
                            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                          >
                             
                            *
                          </span>
                        </label>
                        <div
                          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
                        >
                          <input
                            aria-invalid="false"
                            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                            required=""
                            type="text"
                            value=""
                          />
                          <fieldset
                            aria-hidden="true"
                            class="PrivateNotchedOutline-root-56 MuiOutlinedInput-notchedOutline"
                          >
                            <legend
                              class="PrivateNotchedOutline-legendLabelled-58"
                            >
                              <span>
                                Serial Number
                                 *
                              </span>
                            </legend>
                          </fieldset>
                        </div>
                        
                      </div>
                      <p
                        class="MuiTypography-root makeStyles-infoText-45 MuiTypography-body1"
                      >
                        Enter terminal serial number
                      </p>
                      <div
                        class="MuiFormGroup-root makeStyles-testMode-46"
                      >
                        <label
                          class="MuiFormControlLabel-root"
                        >
                          <span
                            aria-disabled="false"
                            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-61 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
                            data-test-id="autoAddressCheckbox"
                          >
                            <span
                              class="MuiIconButton-label"
                            >
                              <input
                                class="PrivateSwitchBase-input-64"
                                data-indeterminate="false"
                                name="testMode"
                                type="checkbox"
                                value=""
                              />
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                                />
                              </svg>
                            </span>
                            <span
                              class="MuiTouchRipple-root"
                            />
                          </span>
                          <span
                            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                          >
                            Test mode
                          </span>
                        </label>
                      </div>
                      <button
                        class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submitBtn-47 MuiButton-containedPrimary"
                        data-test-id="autoAddressSubmitButton"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiButton-label"
                        >
                          Test Auto address
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                  
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root makeStyles-panelRow-19 MuiGrid-item"
            />
          </div>
        </div>
      </div>
      <div
        aria-labelledby="simple-tab-1"
        hidden=""
        id="simple-tabpanel-1"
        role="tabpanel"
      />
    </div>
  </div>
</div>
`;
